### PR TITLE
[synthetics] Document that not all Playwright functionality is available in Synthetics

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Closes https://github.com/elastic/observability-docs/issues/2603

## Approach

After some discussion in #2603, I've approached this more like _documenting the relationship between Synthetics and Playwright_ rather than just documenting that not all Playwright functionality is available in Synthetics. Here's what I've done so far:

* Added `expect` to the existing [Syntax overview](https://observability-docs_2792.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html#synthetics-syntax) section to highlight that this functionality is built into Synthetics. 
* Added a new [Playwright syntax](https://observability-docs_2792.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html#synthetics-playwright) section to the existing [Add steps](https://observability-docs_2792.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html#synthetics-create-step) section. This new section:
  * Explains the relationship between Synthetics and Playwright using a few examples of commonly used Playwright capabilities (for finding elements on a page and simulating user interaction).
  * Calls out Playwright capabilities that are definitely _not_ supported in Synthetics (for example, videos).
  * Calls out functionality that is available in Playwright, but is not supported/recommended when using Playwright with Synthetics because there is an alternative built into Synthetics (for example, `expect` and `request`).
* Split the content on using `expect` in a `step` into its own section, [Make assertions](https://observability-docs_2792.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html#synthetics-make-assertions) in order to provide more detail.
  * The current draft also includes a reference of all the `expect` methods, but I'm not sure if this is necessary based on the discussion in [#2603](https://github.com/elastic/observability-docs/issues/2603) (since this information could be included in a future generated reference). I'm interested in @vigneshshanmugam and @paulb-elastic's thoughts on this.

## To do

- [x] Get feedback on the approach from @vigneshshanmugam and @paulb-elastic
- [x] If needed, fill in missing `Description...` placeholders
- [ ] Full review from @vigneshshanmugam and/or @paulb-elastic
- [ ] Address feedback and publish

### CODE REVIEW

**Comment by paulb-elastic** [comment]
@colleenmcginnis the build seems to have finished but the [links](https://github.com/elastic/observability-docs/pull/2792#issuecomment-1483248724) aren't working (which would be useful in this context), thanks.

**Comment by colleenmcginnis** [comment]
>@colleenmcginnis the build seems to have finished but the https://github.com/elastic/observability-docs/pull/2792#issuecomment-1483248724 aren't working (which would be useful in this context), thanks.

We're experiencing some issues with the preview generation process. I'll let you know when it's back up!

**Comment by colleenmcginnis** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by colleenmcginnis** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by colleenmcginnis** [comment]
@paulb-elastic all the preview links in the original post should work now! Let me know if you have any issues.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:227]: We use the expect library from the Jest framework which itself is a standalone package. We could also point users to the website - https://jestjs.io/docs/expect. TLDR: Playwright used the same library before and recently enhanced it to support custom assertions which we clearly dont support leading to the whole confusion. 

One more idea would be to actually include only the subset of methods that we support and link the docs directly to the expect library instead. 

Thoughts?

Comment (colleenmcginnis) [reply docs/en/observability/synthetics-create-test.asciidoc:227]: >We could also point users to the website - https://jestjs.io/docs/expect. 

At first glance it looks like not all the methods listed in the Jest docs are included in [`expect.ts`](https://github.com/elastic/synthetics/blob/main/src/core/expect.ts). Does that mean not all the Jest methods are supported in Synthetics? If so, could linking to the Jest site just cause the same kind of confusion we're seeing with linking to Playwright assertions? 

>One more idea would be to actually include only the subset of methods that we support and link the docs directly to the expect library instead.

When you say "link the docs directly to the expect library" do you mean the Jest site or somewhere else?

Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-create-test.asciidoc:227]: > If so, could linking to the Jest site just cause the same kind of confusion we're seeing with linking to Playwright assertions?

Yeah thats a possibility, We didnt expose that as some of the matchers in expect does not make sense for Synthetics. 

> When you say "link the docs directly to the expect library" do you mean the Jest site or somewhere else?

Yeah directly to the jest site for each supported methods. 

I don't have any preference on whether we should add them here vs referencing the users directly there and clearly documentating that we support only partial of them. Thoughts @paulb-elastic @andrewvc

Comment (colleenmcginnis) [reply docs/en/observability/synthetics-create-test.asciidoc:227]: >I don't have any preference on whether we should add them here vs referencing the users directly there and clearly documentating that we support only partial of them.

Curious what Paul and Andrew think, but I'm thinking a full list of the `expect` methods that are supported in Synthetics where each list item is just a link to the relevant part of the Jest docs makes sense to me. That seems like a good compromise -- not too much content that might become difficult to maintain, but also clarity on what is/isn't supported.

Comment (paulb-elastic) [reply docs/en/observability/synthetics-create-test.asciidoc:227]: @colleenmcginnis I 100% agree, I think this is a good compromise.
I am concerned just linking won't be clear exactly what is and isn't supported, or expected to work.  I can see this being a useful reference for us/Support as well as for users.

**Comment by colleenmcginnis** [comment]
@paulb-elastic @vigneshshanmugam I updated the preview based on the conversation above. Would one (or both) of you be able to do a full review? Here's the [updated preview](https://observability-docs_2792.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-create-test.html). Thanks!

**Review by paulb-elastic — APPROVED** [review]
Thanks @colleenmcginnis this looks great, I'll LGTM, but defer to @vigneshshanmugam to get the final approval before merging.

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:29]: Typo 🙈  

(Reminder to myself to fix before merging.)

```suggestion
`expect`::      Check that a value meets a specific condition. There are several supported checks.
                Learn more in <<synthetics-make-assertions>>.
```

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:174]: We might need to mention it as `Playwright functionality that is not supported out of the box`, as users can still access these by using different means via Synthetics ex: video and API request.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM @colleenmcginnis 🎉 

can we update the syntax example in the Make assertions section to be
```ts
  const input = await page.locator('input.new-todo');
    expect(await input.getAttribute('placeholder')).toBe(
      'What needs to be done?'
    );
```

File: docs/en/observability/synthetics-create-test.asciidoc
Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-create-test.asciidoc:188]: ++ I feel this is better than original wording and clearly mentions alternatives and what are not supported out of the box.

**Pull request merged**
